receiver/k8seventsreceiver/k8s_event_to_logdata.go

```diff
@@ -0,0 +1,93 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8seventsreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8seventsreceiver"
+
+import (
+	"strings"
+
+	"go.opentelemetry.io/collector/model/pdata"
+	semconv "go.opentelemetry.io/collector/model/semconv/v1.5.0"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	// Number of log attributes to add to the pdata.LogSlice.
+	totalLogAttributes = 8
+
+	// Number of resource attributes to add to the pdata.ResourceLogs.
+	totalResourceAttributes = 6
+)
+
+// Only two types of events are created as of now.
+// For more info: https://docs.openshift.com/container-platform/4.9/rest_api/metadata_apis/event-core-v1.html
+var severityMap = map[string]pdata.SeverityNumber{
+	"normal":  pdata.SeverityNumberINFO,
+	"warning": pdata.SeverityNumberWARN,
+}
+
+// k8sEventToLogRecord converts Kubernetes event to pdata.LogSlice and adds the resource attributes.
+func k8sEventToLogData(logger *zap.Logger, ev *corev1.Event) pdata.Logs {
+	ld := pdata.NewLogs()
+	rl := ld.ResourceLogs().AppendEmpty()
+	ill := rl.InstrumentationLibraryLogs().AppendEmpty()
+	lr := ill.Logs().AppendEmpty()
+
+	resourceAttrs := rl.Resource().Attributes()
+	resourceAttrs.EnsureCapacity(totalResourceAttributes)
+
+	resourceAttrs.InsertString(semconv.AttributeK8SClusterName, ev.ObjectMeta.ClusterName)
+	resourceAttrs.InsertString("k8s.event.source", ev.Source.Component)
+	resourceAttrs.InsertString(semconv.AttributeK8SNodeName, ev.Source.Host)
+
+	// Attributes related to the object causing the event.
+	resourceAttrs.InsertString("k8s.object.kind", ev.InvolvedObject.Kind)
+	resourceAttrs.InsertString("k8s.object.name", ev.InvolvedObject.Name)
+	resourceAttrs.InsertString("k8s.object.uid", string(ev.InvolvedObject.UID))
+
+	lr.SetTimestamp(pdata.Timestamp(getEventTimestamp(ev).UnixNano()))
+
+	// The Message field contains description about the event,
+	// which is best suited for the "Body" of the LogSlice.
+	lr.Body().SetStringVal(ev.Message)
+
+	// Set the "SeverityNumber" and "SeverityText" if a known type of
+	// severity is found.
+	if severityNumber, ok := severityMap[strings.ToLower(ev.Type)]; ok {
+		lr.SetSeverityNumber(severityNumber)
+		lr.SetSeverityText(ev.Type)
+	} else {
+		logger.Debug("unknown severity type", zap.String("type", ev.Type))
+	}
+
+	attrs := lr.Attributes()
+	attrs.EnsureCapacity(totalLogAttributes)
+
+	attrs.InsertString("k8s.event.reason", ev.Reason)
+	attrs.InsertString("k8s.event.action", ev.Action)
+	attrs.InsertString("k8s.event.start_time", ev.ObjectMeta.CreationTimestamp.String())
+	attrs.InsertString("k8s.event.name", ev.ObjectMeta.Name)
+	attrs.InsertString("k8s.event.uid", string(ev.ObjectMeta.UID))
+	attrs.InsertString("k8s.object.fieldpath", ev.InvolvedObject.FieldPath)
```

Comment:
I don't think this should be a log attribute while others are resource attributes.

Reply:
I was under the impression that the [resource attributes](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-resource) can only be the fields that will have the same value across the time if emitted from the same source, for example, multiple events emitted from the same pod will contain the same pod `name` and `id` but it can contain different container name under the `fieldpath` field.

Please provide your views if I am missing something.

Reply:
I'm thinking of `k8s.object.fieldpath` as an entity that represents a part of a k8s object and that part is assigned to an event. So I think that event's resource can be that part. I don't see any problem with that.

> multiple events emitted from the same pod will contain the same pod `name` and `id` but it can contain different container name under the `fieldpath` field.

Right, and multiple events can be emitted from the container (fieldpath) as well. Or they can also be emitted from different pods.

Reply:
Okay, got your point.
I have made the changes accordingly.